File: src/main/scala/net/machinemuse/powersuits/client/render/modelspec/ArmorMorphTarget.scala
package net.machinemuse.powersuits.client.render.modelspec

import net.minecraft.client.model.{ModelBiped, ModelRenderer}
import org.lwjgl.opengl.GL11._

/**
 * Author: MachineMuse (Claire Semple)
 * Created: 2:09 AM, 29/04/13
 */

sealed trait MorphTarget {
  val name: String
  val degrad = 180F / Math.PI.asInstanceOf[Float]

  def apply(m: ModelBiped): ModelRenderer
}

//case object Cloak extends MorphTarget {
//  val name = "Cloak"
//
//  def apply(m: ModelBiped) = m.bipedCloak
//}

case object Head extends MorphTarget {
  val name = "Head"

  def apply(m: ModelBiped) = m.bipedHead
}

case object Body extends MorphTarget {
  val name = "Body"

  def apply(m: ModelBiped) = m.bipedBody
}

case object RightArm extends MorphTarget {
  val name = "RightArm"

  def apply(m: ModelBiped) = m.bipedRightArm
}

case object LeftArm extends MorphTarget {
  val name = "LeftArm"

  def apply(m: ModelBiped) = m.bipedLeftArm
}

case object RightLeg extends MorphTarget {
  val name = "RightLeg"

  def apply(m: ModelBiped) = m.bipedRightLeg
}

case object LeftLeg extends MorphTarget {
  val name = "LeftLeg"

  def apply(m: ModelBiped) = m.bipedLeftLeg
}

File: src/main/scala/net/machinemuse/powersuits/client/render/modelspec/ArmorMorphTarget.java
//package net.machinemuse.powersuits.client.render.modelspec;
//
//import net.minecraft.client.model.ModelBiped;
//import net.minecraft.client.model.ModelRenderer;
//
///**
// * Ported to Java by lehjr on 11/5/16.
// */
//public class ArmorMorphTarget {
//    public interface MorphTarget {
//        String name = null;
//        float degrad = (float) (180F / Math.PI);
//
//        ModelRenderer apply(ModelBiped m);
//    }
//
//    public static class Cloak implements MorphTarget {
//        String name = "Cloak";
//
//        public ModelRenderer apply(ModelBiped m) {
//            return m.bipedCloak;
//        }
//    }
//
//    public static class Head implements MorphTarget {
//        String name = "Head";
//
//        public ModelRenderer apply(ModelBiped m) {
//            return m.bipedHead;
//        }
//    }
//
//    public static class Body implements MorphTarget {
//        String name = "Body";
//
//        public ModelRenderer apply(ModelBiped m) {
//            return m.bipedBody;
//        }
//    }
//
//    public static class RightArm implements MorphTarget {
//        String name = "RightArm";
//
//        public ModelRenderer apply(ModelBiped m) {
//            return m.bipedRightArm;
//        }
//    }
//
//    public static class LeftArm implements MorphTarget {
//        String name = "LeftArm";
//
//        public ModelRenderer apply(ModelBiped m) {
//            return m.bipedLeftArm;
//        }
//    }
//
//    public static class RightLeg implements MorphTarget {
//        String name = "RightLeg";
//
//        public ModelRenderer apply(ModelBiped m) {
//            return m.bipedRightLeg;
//        }
//    }
//
//    public static class LeftLeg implements MorphTarget {
//        String name = "LeftLeg";
//
//        public ModelRenderer apply(ModelBiped m) {
//            return m.bipedLeftLeg;
//        }
//    }
//}